test/conformance/header_test.go

```diff
@@ -0,0 +1,91 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/serving/test"
+)
+
+// TestMustHaveHeadersSet verified that all headers declared as "MUST" in the runtime
+// contract are present from the point of view of the user container.
+func TestMustHaveHeadersSet(t *testing.T) {
+	logger := logging.GetContextLogger(t.Name())
+	clients := setup(t)
+
+	ri, err := fetchRuntimeInfo(clients, logger, &test.Options{})
+	if err != nil {
+		t.Fatalf("Error fetching runtime info: %v", err)
+	}
+
+	// For incoming requests, the Host header is promoted to the
+	// Request.Host field and removed from the Header map. Therefore we
+	// check against the Host field instead of the map.
+	if ri.Request.Host == "" {
+		// We just check that the host header exists and is non-empty as the request
+		// may be made internally or externally which will result in a different host.
+		t.Error("Header host was not present on request")
+	}
+}
+
+// TestMustHaveHeadersSet verified that all headers declared as "SHOULD" in the runtime
+// contract are present from the point of view of the user container.
+func TestShouldHaveHeadersSet(t *testing.T) {
+	logger := logging.GetContextLogger(t.Name())
+	clients := setup(t)
+
+	expectedHeaders := map[string]*regexp.Regexp{
+		// We expect the protocol to be http for our test image.
+		"x-forwarded-proto": regexp.MustCompile("http"),
```

Comment:
Would you expect this conformance test to pass on a cluster with TLS enabled?

Reply:
Updating to 'https?'